src/auth/integration-tests/src/lib.rs

```diff
@@ -525,3 +525,48 @@ impl SubjectTokenProvider for TestSubjectTokenProvider {
         Ok(SubjectTokenBuilder::new(self.subject_token.clone()).build())
     }
 }
+
+#[cfg(google_cloud_unstable_id_token)]
+pub mod unstable {
+    use auth::credentials::mds::idtoken::Builder as IDTokenMDSBuilder;
+    use base64::{Engine as _, engine::general_purpose::URL_SAFE_NO_PAD};
+    use serde_json::Value;
+
+    pub async fn mds_id_token() -> anyhow::Result<()> {
+        let audience = "https://example.com";
+
+        // Get the service account email from the metadata server directly
+        let client = reqwest::Client::new();
+        let expected_email = client
+            .get("http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/email")
+            .header("Metadata-Flavor", "Google")
+            .send()
+            .await
+            .expect("failed to get service account email from metadata server")
+            .text()
+            .await
+            .expect("failed to read service account email from metadata server response");
+
+        // Only works when running on an env that has MDS.
```

Comment:
I don't like this.

Reply:
me neither haha had to adapt since I came back working on auth stuff, since for building auth features, a lot of them have MDS related features.